app/controllers/services_controller.rb

```diff
@@ -5,9 +5,30 @@ def index
 
   def show
     @service = Service.find_by(slug: params[:service_slug])
-
     @local_authorities = @service.local_authorities.order(name: :asc)
+    @link_count = links_for_service.count
+    @link_filter = params[:filter]
+    @links = links_for_service.group_by(&:local_authority_id)
+  end
+
+private
+
+  def links_for_service
+    @_links_for_service ||= @service
+      .links
+      .where(link_filter_sql)
+      .includes([:service, :interaction, :local_authority])
+      .all
+  end
 
-    @links = @service.links.includes([:service, :interaction, :local_authority]).all.group_by(&:local_authority_id)
+  def link_filter_sql
+    case params[:filter]
+    when 'broken_links'
```

Comment:
Can we use the `currently_broken` scope in the Link model instead, as we've already defined there what a broken link is?  If so, we could also create a scope for 'good links'.